pkg/shell/machine-dialogs.js

```diff
@@ -417,6 +417,9 @@ define([
 
             dialog.run(try_to_connect(dialog.address), function (ex) {
                 if (ex.problem == "no-host") {
+                    if (dialog.address.indexOf(":") == -1)
+                        dialog.address += ":22";
```

Comment:
Can we not modify dialog address here? We aren't going to want that port everywhere it is used. Better to assign to another variable.